packages/eas-cli/src/commands/update/index.ts

```diff
@@ -685,15 +671,17 @@ export default class UpdatePublish extends EasCommand {
       inputDir: flags['input-dir'],
       skipBundler,
       clearCache: flags['clear-cache'] ? true : !!flags['with-eas-environment-variables-set'],
-      platform: flags.platform as RequestedPlatform,
+      platform: flags.platform,
       privateKeyPath: flags['private-key-path'],
       rolloutPercentage: flags['rollout-percentage'],
       nonInteractive,
       emitMetadata,
       json: flags.json ?? false,
-      withEasEnvironmentVariablesSet: flags[
-        'with-eas-environment-variables-set'
-      ] as EnvironmentVariableEnvironment,
+      withEasEnvironmentVariablesSet: flags['with-eas-environment-variables-set']
+        ? isEnvironment(flags['with-eas-environment-variables-set'])
+          ? flags['with-eas-environment-variables-set']
+          : null
+        : null,
```

Comment:
```suggestion
      withEasEnvironmentVariablesSet: flags['with-eas-environment-variables-set'],
```
(since this is checked above on L664)

Reply:
It totally makes sense, but TS doesn't seem to respect the checks done above, and to avoid typecasting I needed to do it here for the second time

Reply:
Here's how I'd structure it (since we'll be using this flag in various places):

```typescript
const withEasEnvironmentVariablesSetFlag = getEnvironmentFlagValue(
      flags['with-eas-environment-variables-set']
    );
```

```typescript
export function getEnvironmentFlagValue(
  maybeEnvironment: string | null | undefined
): EnvironmentVariableEnvironment | null | undefined {
  if (maybeEnvironment === null || maybeEnvironment === undefined) {
    return maybeEnvironment;
  }

  if (!isEnvironment(maybeEnvironment)) {
    Errors.error(
      `--with-eas-environment-variables-set must be one of ${Object.values(
        EnvironmentVariableEnvironment
      )
        .map(env => `"${env.toLocaleLowerCase()}"`)
        .join(', ')}`,
      { exit: 1 }
    );
  }

  return maybeEnvironment;
}
```

```typescript
withEasEnvironmentVariablesSet: withEasEnvironmentVariablesSetFlag ?? null,
```